### ISSUE TITLE
security exceptions in build

### ISSUE THREAD

**Issue by dhruba** [dropbox/djinni#22]
I am trying to compile Djinni, but am running into some Java security exceptions. If anybody has any suggestions on how to switch off these security settings, that would be great.

dhruba@dev223 ~/djinni]$ java -version
java version "1.7.0_06"
Java(TM) SE Runtime Environment (build 1.7.0_06-b24)
Java HotSpot(TM) 64-Bit Server VM (build 23.2-b09, mixed mode)

[dhruba@dev223 ~/djinni]$ uname -a
Linux 3.2.55-106_fbk22_00877_g6902630 #106 SMP Wed Apr 2 16:19:59 PDT 2014 x86_64 x86_64 x86_64 GNU/Linux

[dhruba@dev223 ~/djinni]$ src/run
Building Djinni...
java -Xms512M -Xmx1024M -Xss1M -XX:+CMSClassUnloadingEnabled -XX:MaxPermSize=256M -Djava.security.manager -Djava.security.policy=/home/dhruba/local/djinni/src/support/sbt.security.policy -jar /home/dhruba/local/djinni/src/support/sbt-launch.jar -Dsbt.override.build.repos=true -Dsbt.repository.config=/home/dhruba/local/djinni/src/support/sbt.resolvers.properties compile start-script
[info] Loading project definition from /data/users/dhruba/djinni/src/project
[info] Updating {file:/data/users/dhruba/djinni/src/project/}src-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: com.typesafe.sbt#sbt-start-script;0.10.0: several problems occurred while resolving dependency: com.typesafe.sbt#sbt-start-script;0.10.0 {compile=[default(compile)]}:
[warn]  java.security.AccessControlException: access denied ("java.net.SocketPermission" "xx:8080" "connect,resolve")
[warn]  java.security.AccessControlException: access denied ("java.net.SocketPermission" "xx:8080" "connect,resolve")
[warn]  java.security.AccessControlException: access denied ("java.net.SocketPermission" "xx:8080" "connect,resolve")
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn]  Note: Some unresolved dependencies have extra attributes.  Check that these dependencies exist with the requested attributes.
[warn]      com.typesafe.sbt:sbt-start-script:0.10.0 (sbtVersion=0.13, scalaVersion=2.10)
[warn] 
sbt.ResolveException: unresolved dependency: com.typesafe.sbt#sbt-start-script;0.10.0: several problems occurred while resolving dependency: com.typesafe.sbt#sbt-start-script;0.10.0 {compile=[default(compile)]}:
    java.security.AccessControlException: access denied ("java.net.SocketPermission" "xx.xx.xx.xx:8080" "connect,resolve")
    java.security.AccessControlException: access denied ("java.net.SocketPermission" "xx.xx.xx.xx:8080" "connect,resolve")
    java.security.AccessControlException: access denied ("java.net.SocketPermission" "xx.xx.xx.xx:8080" "connect,resolve")
    at sbt.IvyActions$.sbt$IvyActions$$resolve(IvyActions.scala:217)
    at sbt.IvyActions$$anonfun$update$1.apply(IvyActions.scala:126)
    at sbt.IvyActions$$anonfun$update$1.apply(IvyActions.scala:125)
    at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:115)
    at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:115)
    at sbt.IvySbt$$anonfun$withIvy$1.apply(Ivy.scala:103)
    at sbt.IvySbt.sbt$IvySbt$$action$1(Ivy.scala:48)
    at sbt.IvySbt$$anon$3.call(Ivy.scala:57)
    at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:98)
    at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:81)
    at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:102)
    at xsbt.boot.Using$.withResource(Using.scala:11)
    at xsbt.boot.Using$.apply(Using.scala:10)
    at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:62)
    at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:52)
    at xsbt.boot.Locks$.apply0(Locks.scala:31)
    at xsbt.boot.Locks$.apply(Locks.scala:28)
    at sbt.IvySbt.withDefaultLogger(Ivy.scala:57)
    at sbt.IvySbt.withIvy(Ivy.scala:98)
    at sbt.IvySbt.withIvy(Ivy.scala:94)
    at sbt.IvySbt$Module.withModule(Ivy.scala:115)
    at sbt.IvyActions$.update(IvyActions.scala:125)
    at sbt.Classpaths$$anonfun$sbt$Classpaths$$work$1$1.apply(Defaults.scala:1223)
    at sbt.Classpaths$$anonfun$sbt$Classpaths$$work$1$1.apply(Defaults.scala:1221)
    at sbt.Classpaths$$anonfun$doWork$1$1$$anonfun$74.apply(Defaults.scala:1244)
    at sbt.Classpaths$$anonfun$doWork$1$1$$anonfun$74.apply(Defaults.scala:1242)
    at sbt.Tracked$$anonfun$lastOutput$1.apply(Tracked.scala:35)
    at sbt.Classpaths$$anonfun$doWork$1$1.apply(Defaults.scala:1246)
    at sbt.Classpaths$$anonfun$doWork$1$1.apply(Defaults.scala:1241)
    at sbt.Tracked$$anonfun$inputChanged$1.apply(Tracked.scala:45)
    at sbt.Classpaths$.cachedUpdate(Defaults.scala:1249)
    at sbt.Classpaths$$anonfun$updateTask$1.apply(Defaults.scala:1214)
    at sbt.Classpaths$$anonfun$updateTask$1.apply(Defaults.scala:1192)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:42)
    at sbt.std.Transform$$anon$4.work(System.scala:64)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:18)
    at sbt.Execute.work(Execute.scala:244)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:160)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:30)
    at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
    at java.util.concurrent.FutureTask.run(FutureTask.java:166)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
    at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
    at java.util.concurrent.FutureTask.run(FutureTask.java:166)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603)
    at java.lang.Thread.run(Thread.java:722)
[error](*:update) sbt.ResolveException: unresolved dependency: com.typesafe.sbt#sbt-start-script;0.10.0: several problems occurred while resolving dependency: com.typesafe.sbt#sbt-start-script;0.10.0 {compile=[default(compile)]}:
[error]     java.security.AccessControlException: access denied ("java.net.SocketPermission" "xx.xx.xx.xx:8080" "connect,resolve")
[error]     java.security.AccessControlException: access denied ("java.net.SocketPermission" "xx.xx.xx.xx:8080" "connect,resolve")
[error]     java.security.AccessControlException: access denied ("java.net.SocketPermission" "xx.xx.xx.xx:8080" "connect,resolve")

**Comment by hamedmohammadi** [comment]
same issue for me on ubuntu ,and it will work with sudo but i'm not sure if it is a good idea

**Comment by j4cbo** [comment]
@cakoose - any idea why this might be giving false positives here?

**Comment by cakoose** [comment]
Earlier, we had issues with SBT using HTTP instead of HTTPS.  We fixed the config, but since the config is sort of subtle and finicky, we added a second line of defense: a security policy file that only allowed connections on port 443 ("src/support/sbt.security.policy").

I haven't looked into it in detail yet, but some things that might work:
- Allow connections based on the actual protocol used (HTTPS) instead of the port (443).
- Somehow specifically allow proxy servers.

Or maybe we can just remove this second line of defense.
